roles/postgres/molecule/default/verify.yml

```diff
@@ -0,0 +1,49 @@
+
+---
+- name: postgres Tests
+  hosts: all
+  become: yes
+  become_user: postgres
+  vars:
+    postgres_major_version: "15"
+  tasks:
+  - name: Gather service facts
+    ansible.builtin.service_facts:
+
+  - name: Set PostgreSQL service name based on OS type
+    ansible.builtin.set_fact:
+      service_name: >-
+        {{ 'postgresql' if ansible_os_family != 'RedHat'
+          else 'postgresql-' ~ postgres_major_version }}
+
+  - name: Fail if PostgreSQL service is not running or enabled
+    ansible.builtin.fail:
+      msg: "PostgreSQL service '{{ service_name }}.service' is not running or enabled!"
+    when: >
+      ansible_facts.services.get(service_name ~ '.service', {}).state != 'running' or
+      ansible_facts.services.get(service_name ~ '.service', {}).status != 'enabled'
```

Comment:
I think using the straightforward logic would be easier.
Check out the assert module to assert facts populated by service_facts (no need to negate stuff here)